rasp/base.py

```diff
@@ -67,9 +67,12 @@ def as_func(cls, method_name):
         try:
             return getattr(TmpEngine, method_name)
         except AttributeError:
-            raise NotImplementedError('Class {} does not implement {}'.format(class_name, func_name))
+            raise NotImplementedError('Class {} does not implement {}'.format(class_name, method_name))
 
-    def get_page_source(self, url, params=None, headers=None):
+    def get_page_source(self, url,
+                        params=None,
+                        headers=None,
+                        pre_fetch_callback=None):
```

Comment:
The `pre_fetch_callback` could (and maybe should) be specified at class init.  If we do that, it will affect how the func/copy operation works, so we will have to take that into account.